catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -151,16 +199,73 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
         return this.attributeOverrides;
     }
 
-    private void configureCamelRoute() {
-        LOGGER.trace("ENTERING: configureCamelRoute");
+    public List<RouteDefinition> getRouteDefinitions() {
+        return camelContext.getRouteDefinitions();
+    }
 
-        // Must have a directory to be monitored to be able to configure the Camel route.
-        if (StringUtils.isEmpty(monitoredDirectory)) {
-            LOGGER.debug("Cannot setup camel route - must specify a directory to be monitored");
-            return;
+    /*
+        Task that waits for the "content" CamelComponent before adding the routes from the
+        RouteBuilder to the CamelContext. This ensures content directory monitors will
+        automatically start after a system shutdown.
+     */
+    private void attemptAddRoutes() {
+        LOGGER.debug("Attempting to add routes for content directory monitor watching {}",
+                monitoredDirectory);
+        RouteBuilder routeBuilder = createRouteBuilder();
+        try {
+            waitForContentCamelComponent();
+            camelContext.addRoutes(routeBuilder);
+            setRouteCollection(routeBuilder);
+        } catch (InterruptedException e) {
+            Thread.currentThread()
+                    .interrupt();
+            LOGGER.debug(
+                    "Content directory monitor interrupted while waiting for Camel component 'content'. ");
+        } catch (Exception e) {
+            LOGGER.warn("{} {}",
+                    "Unable to configure Camel route.",
+                    "This content directory monitor will be unusable.");
+        } finally {
+            if (LOGGER.isDebugEnabled()) {
+                dumpCamelContext("after configureCamelRoute()");
+            }
         }
+    }
+
+    /*
+        Assign this content directory monitor's routeCollection to the routes generated by
+        this content directory monitor's RouteBuilder.
+     */
+    private void setRouteCollection(RouteBuilder routeBuilder) {
+        routeCollection = routeBuilder.getRouteCollection()
+                .getRoutes();
+    }
 
-        RouteBuilder routeBuilder = new RouteBuilder() {
+    /*
+        Do not attempt to add routes to the CamelContext until we know the content scheme is ready.
+     */
+    private void waitForContentCamelComponent() throws InterruptedException {
+        Failsafe.with(retryPolicyForContent)
+                .run(executionContext -> {
```

Comment:
Can you try `retryWhen` and `retryIf` using `get` instead of `run`?  You should not have to throw an exception to trigger a retry.